src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -1436,45 +1434,53 @@ _mongoc_topology_pop_server_session (mongoc_topology_t *topology,
    int64_t timeout;
    mongoc_server_session_t *ss = NULL;
    mongoc_topology_description_t *td;
+   bool loadbalanced;
 
    ENTRY;
 
    bson_mutex_lock (&topology->mutex);
 
    td = &topology->description;
    timeout = td->session_timeout_minutes;
+   loadbalanced = td->type == MONGOC_TOPOLOGY_LOAD_BALANCED;
 
-   if (timeout == MONGOC_NO_SESSIONS) {
-      /* if needed, connect and check for session timeout again */
-      if (!mongoc_topology_description_has_data_node (td)) {
-         bson_mutex_unlock (&topology->mutex);
-         if (!mongoc_topology_select_server_id (
-                topology, MONGOC_SS_READ, NULL, error)) {
-            RETURN (NULL);
-         }
+   /* When the topology type is LoadBalanced, sessions are always supported. */
+   if (!loadbalanced) {
+      if (timeout == MONGOC_NO_SESSIONS) {
```

Comment:
Since we have two nested `if`s and no nested `else`, would it make sense to just change the condition here and minimize the diff?

```c
if (!loadbalanced && timeout == MONGOC_NO_SESSIONS) {
```